pkg/kubernetes/manifest/manifest.go

```diff
@@ -145,6 +145,13 @@ func (m Metadata) Labels() map[string]interface{} {
 	return m["labels"].(map[string]interface{})
 }
 
+func (m Metadata) SetLabel(key, value string) {
```

Comment:
You don't need that, `m.Labels()` already makes sure it returns a valid `map[string]interface{}`. Again for modifying maps, I prefer to do it using the Go bracket syntax, because it makes clear that you modify the pointer map there.

Reply:
m.Labels() does return a new map, but it does not assign it back into `m["labels"]`, so changes don't stick.
At first I just changed `Labels()` to do `m["labels"] = make....`, but I didn't want it to create an empty map simply because something wanted to read the collection.

But if we are going toward the "just do it" approach to labelling, that is not a concern.

Reply:
Oh, I mixed up code in my head, I changed it in my PR so it creates AND returns if nil. I'd go forward like so

Reply:
Basically `nil` labels is a thing related to the Go type system, but logically equivalent to an empty map. So it's fine to initiate and return one in case